### PR TITLE
Lantern mobile settings

### PR DESCRIPTION
**Pull request by atavism**
- When we can't find ``settings.yaml``, always start with a default configuration
- Move sdk directory underneath app project with symlink for easier reference
- Include recent updates to Tun2Socks
- Now that we're extending AppCompatActivity again, hide the default action bar given since we're using a custom design
- Build a new version of the app with Go 1.6 and latest gomobile

### CODE REVIEW

File: src/github.com/getlantern/lantern-mobile/Makefile [hunk 1]
```diff
@@ -1,77 +0,0 @@
-S3CMD := $(shell which s3cmd 2> /dev/null)
```
Comment (oxtoacart) [inline src/github.com/getlantern/lantern-mobile/Makefile]: @atavism Why did you get rid of the Makefile?

**Comment by oxtoacart** [comment]
@atavism I'm having trouble building.  I tried the following:

1. Run make android-lib in root of project
2. Then in src/github.com/getlantern/lantern-mobile I ran "gradle assembleDebug"

It complains of this:

```
/Users/ox.to.a.cart/lantern2/src/github.com/getlantern/lantern-mobile/sdk/src/main/java/org/getlantern/lantern/sdk/Lantern.java:17: error: package go.client does not exist
import go.client.*;
```

File: src/github.com/getlantern/lantern-mobile/Makefile (hunk 1 above)
Comment (atavism) [inline src/github.com/getlantern/lantern-mobile/Makefile]: We don't use this Makefile anymore so I figured I'd delete it.

**Comment by atavism** [comment]
What about just doing "make android-lib android" from the root of the project? That's how I build the app.

**Comment by oxtoacart** [comment]
@atavism That worked sort of.  But it's complaining about not being able to load tun2socks:

```
         AndroidRuntime  E  FATAL EXCEPTION: Thread-11432
                         E  Process: org.getlantern.lantern, PID: 20548
                         E  java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader[DexPathList[[zip file "/data/app/org.getlantern.lantern-1/base.apk"],nativeLibraryDirectories=[/dat
                            a/app/org.getlantern.lantern-1/lib/arm, /vendor/lib, /system/lib]]] couldn't find "libtun2socks.so"
                         E      at java.lang.Runtime.loadLibrary(Runtime.java:366)
                         E      at java.lang.System.loadLibrary(System.java:988)
                         E      at org.getlantern.lantern.android.vpn.Tun2Socks.<clinit>(Tun2Socks.java:127)
                         E      at org.getlantern.lantern.vpn.VpnBuilder$1.run(VpnBuilder.java:111)
```

When I try to `make build-tun2socks`, I get the below:

```
➜  lantern2 git:(lantern-mobile-settings) ✗ make build-tun2socks  android
cd src/github.com/getlantern/lantern-mobile
ndk-build
Android NDK: Could not find application project directory !
Android NDK: Please define the NDK_PROJECT_PATH variable to point to it.
make[1]: Entering directory `/Users/ox.to.a.cart/lantern2'
/opt/android-ndk-r10e/build/core/build-local.mk:143: *** Android NDK: Aborting    .  Stop.
make[1]: Leaving directory `/Users/ox.to.a.cart/lantern2'
make: *** [build-tun2socks] Error 2
```

**Comment by atavism** [comment]
Ok, I'll fix that. Can you run ndk-build from the lantern-mobile directory?

**Comment by oxtoacart** [comment]
`ndk-build` works from inside lantern-mobile, but if I Then run `make android-lib android`, I still get the same crash as if it's not picking up the tun2socks library.

**Comment by atavism** [comment]
Thanks, ``build-tun2socks`` is a dependency now, and I made sure we're copying ``libtun2socks.so`` over to the app's ``libs/armeabi-v7a`` directory. Want to try again?

**Comment by oxtoacart** [comment]
Now I get this:

```
make[1]: Leaving directory `/Users/ox.to.a.cart/lantern2/src/github.com/getlantern/lantern-mobile'
cp src/github.com/getlantern/lantern-mobile/libs/armeabi-v7a/libtun2socks.so src/github.com/getlantern/lantern-mobile/app/libs/armeabi-v7a/libtun2socks.so
cp: src/github.com/getlantern/lantern-mobile/app/libs/armeabi-v7a/libtun2socks.so: No such file or directory
```

**Comment by atavism** [comment]
@oxtoacart, added a mkdir before we copy the library

**Comment by oxtoacart** [comment]
Okay, it builds now and turns on with crashing.  Here are the results from testing on my Nexus 4 running 5.1.1.

1. Was able to turn on and confirm that it was proxying traffic from Chrome
2. After turning off, pidcat reported the lantern process crashing with a panic

```
                     Go  E  fatal error: concurrent map writes
                         E  goroutine 75 [running]:
                         E  runtime.throw(0xa11aa358, 0x15)
                         E      /usr/local/go/src/runtime/panic.go:530 +0x80 fp=0x9071be58 sp=0x9071be4c
                         E  runtime.mapassign1(0xa10766d8, 0x90887a80, 0x90756e08, 0x9071bec8)
                         E      /usr/local/go/src/runtime/hashmap.go:445 +0xa0 fp=0x9071beac sp=0x9071be58
                         E  github.com/getlantern/lantern-mobile/lantern/interceptor.(*Interceptor).closeConn(0xa13fad00, 0x90756e00)
                         E      /Users/ox.to.a.cart/lantern2/src/github.com/getlantern/lantern-mobile/lantern/interceptor/interceptor.go:290 +0x1b0 fp=0x9071bee8 sp=0x9071beac
                         E  github.com/getlantern/lantern-mobile/lantern/interceptor.(*Interceptor).Do(0xa13fad00, 0x90b504b0, 0x0, 0x0)
                         E      /Users/ox.to.a.cart/lantern2/src/github.com/getlantern/lantern-mobile/lantern/interceptor/interceptor.go:211 +0x718 fp=0x9071bf94 sp=0x9071bee8
                         E  github.com/getlantern/lantern-mobile/lantern/interceptor.(*Interceptor).acceptSocks.func1(0xa13fad00, 0x90b504b0)
                         E      /Users/ox.to.a.cart/lantern2/src/github.com/getlantern/lantern-mobile/lantern/interceptor/interceptor.go:139 +0x24 fp=0x9071bfcc sp=0x9071bf94
                         E  runtime.goexit()
                         E      /usr/local/go/src/runtime/asm_arm.s:986 +0x4 fp=0x9071bfcc sp=0x9071bfcc
```

3. After turning back on, proxying from Chrome worked
4. After turning off a 2nd time, the Lantern process hung on shutting down (debug output below).  At this point, Chrome was unable to connect to `whatismyipaddress.com`.  I had to manually disable the VPN to restore connectivity, at which point the Lantern process ended.

```
                Lantern  D  About to stop Lantern..
                  GoLog  I  DEBUG lantern: lantern.go:273 Calling exit func
                         I  DEBUG lantern: lantern.go:273 Calling exit func
                         I  DEBUG lantern: lantern.go:273 Calling exit func
                         I  DEBUG lantern: lantern.go:273 Calling exit func
                         I  DEBUG lantern: lantern.go:273 Calling exit func
                         I  DEBUG lantern: lantern.go:273 Calling exit func
                         I  DEBUG flashlight.config: config.go:271 Closing config system
                         I  DEBUG lantern: lantern.go:276 No exit func remaining, exit now
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57616:52.34.67.97:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57638:74.125.224.24:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57646:64.233.189.95:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57604:74.125.23.157:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57601:104.101.209.19:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57605:74.125.23.157:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57645:64.233.189.113:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57652:104.101.209.19:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57655:173.194.72.156:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57580:127.0.0.1:7300
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57602:104.101.209.19:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57603:104.101.209.19:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57601:104.101.209.19:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57606:74.125.23.157:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57616:52.34.67.97:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57615:52.34.67.97:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57617:64.233.189.101:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57638:74.125.224.24:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57619:64.233.189.101:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57604:74.125.23.157:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57646:64.233.189.95:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57659:216.58.221.132:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57605:74.125.23.157:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57600:23.197.144.45:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57595:64.233.189.95:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57645:64.233.189.113:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57599:64.233.189.95:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57652:104.101.209.19:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57607:74.125.23.157:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57614:52.34.67.97:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57655:173.194.72.156:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57580:127.0.0.1:7300
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57602:104.101.209.19:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57600:23.197.144.45:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57619:64.233.189.101:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57618:64.233.189.101:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57659:216.58.221.132:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57641:74.125.23.156:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57595:64.233.189.95:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57660:64.233.189.132:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57599:64.233.189.95:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:162 Closing a connection with id: 127.0.0.1:57593:64.233.189.95:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57607:74.125.23.157:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57614:52.34.67.97:80
                         I  DEBUG lantern: lantern.go:276 No exit func remaining, exit now
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57641:74.125.23.156:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57660:64.233.189.132:443
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57618:64.233.189.101:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57603:104.101.209.19:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57615:52.34.67.97:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57593:64.233.189.95:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57617:64.233.189.101:80
                         I  DEBUG lantern-android.interceptor: interceptor.go:283 Closing a connection with id: 127.0.0.1:57606:74.125.23.157:80
```

**Comment by atavism** [comment]
Thanks, @oxtoacart. I'm able to reproduce that "concurrent map writes" crash after a few tries. I've never encountered it before. I wonder if its related to upgrading to Go 1.6 and/or the latest gomobile. 

I also found a second bug where Lantern, occasionally, mistakenly exits immediately after the first start attempt. Looking into both of these and will have more for you to review tomorrow.

**Comment by oxtoacart** [comment]
For the concurrent map writes thing, you can probably just use a sync.Mutex to guard access.

**Comment by atavism** [comment]
Right, on the concurrent map writes issue. It just never popped up as a fatal error before. 

I'm actually going to close this PR as I feel a lot of changes I wanted to (and started to) make on this branch will be superseded by the work being done in valencia-embeddable anyway. I'll just wait for that to get merged without duplicating work.

**Pull request closed without merging**